ananse/influence.py

```diff
@@ -33,41 +33,165 @@
+        # all the extra data is stored in edge attributes
+        if edge_atribute_length == 6:
+            tf_expression = (
+                0 if len(row) < edge_atribute_length else float(row["tf_expression"])
+            )
+            target_expression = (
+                0
+                if len(row) < edge_atribute_length
+                else float(row["target_expression"])
+            )
+            weighted_binding = (
+                0 if len(row) < edge_atribute_length else float(row["weighted_binding"])
+            )
+            activity = 0 if len(row) < edge_atribute_length else float(row["activity"])
+            try:
+                G.add_edge(
+                    source,
+                    target,
+                    weight=weight,
+                    weighted_binding=weighted_binding,
+                    TF_expression=tf_expression,
+                    TF_activity=activity,
+                    TG_expression=target_expression,
+                )
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
+
+        # if there is only prob score, only that is inc in the GRN
+        else:
+            try:
+                G.add_edge(source, target, weight=weight, n=1)
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
     return G
 
 
-def difference(S, R):
-    """Calculate the network different between two cell types."""
-    DIF = nx.create_empty_copy(R)
-    for (u, v, d) in S.edges(data=True):
-        if (u, v) not in R.edges:
-            DIF.add_edge(u, v, weight=d["weight"], n=1)
-        else:
-            diff_weight = S.edges[u, v]["weight"] - R.edges[u, v]["weight"]
-            if diff_weight > 0:
+def difference(GRN_source, GRN_target):
+    """Calculate the network different between two GRNs.
+    It first takes the nodes from both networks, and subsequently
+    First adds the edges from the target network that are missing in
+    the source network.
+    Secondly add the edges present in both but with a higher interaction
+    score in the target network
+    """
+    DIF = nx.create_empty_copy(GRN_source)  # copying source GRN nodes
+    nodes_target = nx.create_empty_copy(GRN_target)  # copying target GRN nodes
+    DIF.add_nodes_from(
+        list(nodes_target.nodes)
+    )  # merge the nodes if there are differences
+    # (which can happen) when taking the header instead of the --unnion-interactions flagg
+
+    # lets check if the full GRN output is loaded and if so output all atributes to the diffnetwork:
+    full_output_GRN = bool(nx.get_edge_attributes(GRN_source, "TF_expression"))
+    if full_output_GRN:
+        logger.info("calculating diff GRN with full output")
+        # lets load all the differential edges into the diffnetwork
+        for (u, v, a) in GRN_target.edges(data=True):
+            if (u, v) not in GRN_source.edges:
```

Comment:
this whole section can be removed if we only use your top-edges-method right?

Reply:
Hmm... I'm not entirely sure. I think there is also a posibility to run Ananse with only one network. However I gues then exporting the --full-output of influence wouldnt make any sence in the first place. I think for now I'll program it that its not an option to run with a single network and use the --full-output flagg simultainously. Because this whole part does drastically decreas readability/understandability.

Reply:
Wait, it crashes anyway with only one network in the current implementation. I'll stop being afraid of breaking it.